### ISSUE TITLE
Crash on index page

### ISSUE THREAD

**Issue by dlsteuer** [StemboltHQ/battle_snake#4]
Steps:
* on a brand new server, create a single player game
* save game
* stop server
* restart server
When the server has restarted here is what I get:
<img width="1419" alt="screen shot 2017-02-09 at 6 20 24 pm" src="https://cloud.githubusercontent.com/assets/326884/22811572/7ab23ed2-eef4-11e6-9d29-019d9e38ba8c.png">

**Comment by Dkendal** [comment]
Thanks for the report @dlsteuer.

Firstly, I apologise for not having the docs up to date for getting the dev server going.

What's going here is that the schema for [Mnesia](erlang.org/doc/man/mnesia.html) has not been initialised for your local server.

I'm going to be making it my focus tomorrow to make the server setup as painless as I can, but in the mean time here's how you can get your server running:

```sh
# Get to the project root...
git clone https://github.com/StemboltHQ/battle_snake
cd battle_snake

# Install npm dependencies
npm install

# Install and compile elixir dependencies
mix do deps.get, compile

# Start the local dev server
./scripts/dev-server

# The above script starts the web app on port 3000
# It also starts it in an interactive prompt (IEx)
# which we will use to issue a few commands
```

```elixir
# After Starting the server it will barf up some information about Mnesia
# it should look something like this
Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

Setting up watches.  Beware: since -r was given, this may take a while!
[info] Running BattleSnake.Endpoint with Cowboy using http://localhost:3000
Watches established.
---> Processes holding locks <--- 
---> Processes waiting for locks <--- 
---> Participant transactions <--- 
---> Coordinator transactions <---
---> Uncertain transactions <--- 
---> Active tables <--- 
schema         : with 1        records occupying 415      words of mem
===> System info in version "4.14.2", debug level = none <===
opt_disc. Directory "/home/dylan/src/battle_snake/Mnesia.battle-snake-dev@localhost" is NOT used.
use fallback at restart = false
running db nodes   = ['battle-snake-dev@localhost']
stopped db nodes   = [] 
master node tables = []
remote             = []
ram_copies         = [schema]
disc_copies        = []
disc_only_copies   = []
[{'battle-snake-dev@localhost',ram_copies}] = [schema]
2 transactions committed, 0 aborted, 0 restarted, 0 logged to disc
0 held locks, 0 in queue; 0 local transactions, 0 remote
0 transactions waits for other nodes: []
Interactive Elixir (1.4.0) - press Ctrl+C to exit (type h() ENTER for help)
iex(battle-snake-dev@localhost)1> 09 Feb 21:26:13 - info: compiled 10 files into 2 files, copied 3 in 1.2 sec
```

```elixir
# This means that your disc based storage does not exist
opt_disc. Directory "/home/dylan/src/battle_snake/Mnesia.battle-snake-dev@localhost" is NOT used.
```

```elixir
# This means there is only one table, the schema, and it only exists in RAM
# E.g. there is no disk based schema
ram_copies         = [schema]
disc_copies        = []
disc_only_copies   = []
```

```elixir
# We can use the running IEx console to setup the
# database now
# ./scripts/dev-server
iex(battle-snake-dev@localhost)1> :stopped = :mnesia.stop
:stopped
[info] Application mnesia exited: :stopped
iex(battle-snake-dev@localhost)2>
iex(battle-snake-dev@localhost)3> :mnesia.create_schema([node()])
:ok
iex(battle-snake-dev@localhost)3> :ok = :mnesia.start()
:ok
iex(battle-snake-dev@localhost)4>
{:atomic, :ok} = BattleSnake.GameForm.create_table(disc_copies: [node()])
iex(battle-snake-dev@localhost)5>
{:atomic, :ok} = BattleSnake.World.create_table(disc_copies: [node()])
iex(battle-snake-dev@localhost)6>
{:atomic, :ok} = :mnesia.add_table_index(BattleSnake.World.table_name(), :game_form_id)

# And we can print out the info again and see the the
# Db is setup properly now.
iex(battle-snake-dev@localhost)7> :mnesia.info()
---> Processes holding locks <--- 
---> Processes waiting for locks <--- 
---> Participant transactions <--- 
---> Coordinator transactions <---
---> Uncertain transactions <--- 
---> Active tables <--- 
Elixir.BattleSnake.World: with 0        records occupying 298      words of mem
Elixir.BattleSnake.GameForm: with 0        records occupying 298      words of mem
schema         : with 3        records occupying 689      words of mem
===> System info in version "4.14.2", debug level = none <===
opt_disc. Directory "/home/dylan/src/battle_snake/Mnesia.battle-snake-dev@localhost" is used.
use fallback at restart = false
running db nodes   = ['battle-snake-dev@localhost']
stopped db nodes   = [] 
master node tables = []
remote             = []
ram_copies         = []
disc_copies        = ['Elixir.BattleSnake.GameForm',
                      'Elixir.BattleSnake.World',schema]
disc_only_copies   = []
[{'battle-snake-dev@localhost',disc_copies}] = [schema,
                                                'Elixir.BattleSnake.GameForm',
                                                'Elixir.BattleSnake.World']
5 transactions committed, 0 aborted, 0 restarted, 6 logged to disc
0 held locks, 0 in queue; 0 local transactions, 0 remote
0 transactions waits for other nodes: []
:ok
```

# TL;DR
Run `./scripts/dev-server`
Then paste this into the IEx prompt

```elixir
:stopped = :mnesia.stop
:ok = :mnesia.create_schema([node()])
:ok = :mnesia.start()
{:atomic, :ok} = BattleSnake.GameForm.create_table(disc_copies: [node()])
{:atomic, :ok} = BattleSnake.World.create_table(disc_copies: [node()])
{:atomic, :ok} = :mnesia.add_table_index(BattleSnake.World.table_name(), :game_form_id)
```
![1486706028](https://cloud.githubusercontent.com/assets/3162299/22815758/93ab93fc-ef12-11e6-9250-c60bf0d85451.png)
![1486706049](https://cloud.githubusercontent.com/assets/3162299/22815760/93ad8518-ef12-11e6-9220-51b7d465a2d9.png)
![1486706070](https://cloud.githubusercontent.com/assets/3162299/22815759/93ac54b8-ef12-11e6-8def-ae3d906043d0.png)
add snake url
![1486706088](https://cloud.githubusercontent.com/assets/3162299/22815761/93b161c4-ef12-11e6-85ad-7d974c00dcef.png)
click play

Keybindings:
`q` : reset game
`j` or `up` resume
`k` or `down` pause
`h` or `left` previous turn
`l` or `right` next turn

![1486706116](https://cloud.githubusercontent.com/assets/3162299/22815762/93b1871c-ef12-11e6-8da7-759f126b64b9.png)

**Comment by dlsteuer** [comment]
cool thanks, it seemed ok, because on the first run everything came up. after these instructions it's running. I'll start testing the server to see how it does.
